# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../strace/mips64-hotfix.patch
# Copyright (C) 2018 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- strace-4.16/linux/mips/arch_sigreturn.c.vanilla	2018-01-14 10:00:14.602114292 +0000
+++ strace-4.16/linux/mips/arch_sigreturn.c	2018-01-14 10:01:25.922111681 +0000
@@ -1,3 +1,4 @@
+
 static void
 arch_sigreturn(struct tcb *tcp)
 {
@@ -17,7 +18,7 @@
 	 * sizeof(sf_ass) + sizeof(sf_pad) + sizeof(struct siginfo)
 	 */
 	const kernel_ulong_t addr = mips_REG_SP + 6 * 4 + 128 +
-				   offsetof(struct ucontext, uc_sigmask);
+				   offsetof(struct ucontext_t, uc_sigmask);
 #endif
 
 	tprints("{mask=");
